File: src/main/java/io/myosotisdev/utopianism/command/GuildCommand.java
package io.myosotisdev.utopianism.command;

import io.myosotisdev.minestom.command.AbstractCommand;
import io.myosotisdev.minestom.permission.Permission;
import io.myosotisdev.utopianism.util.Namespaces;

public class GuildCommand
        extends AbstractCommand
{
    public static final String   Name    = "guild";
    public static final String[] Aliases = new String[0];

    public GuildCommand(AbstractCommand command)
    {
        super(command, Name, Aliases);

        addSyntax((sender, context) -> {

        });

        addSubcommand(new Create(this));
    }

    @Override
    public Permission permission()
    {
        return Permission.ofString(Namespaces.Utopianism, "guild");
    }

    class Create
            extends AbstractCommand
    {
        public static final String Name = "create";

        public Create(AbstractCommand command)
        {
            super(command, Name);
        }

        @Override
        public Permission permission()
        {
            return Permission.ofString(Namespaces.Utopianism, "guild.create");
        }
    }

    class Dissolve extends AbstractCommand
    {
        public static final String Name = "dissolve";

        public Dissolve(AbstractCommand command)
        {
            super(command, Name);
        }

        @Override
        public Permission permission()
        {
            return Permission.ofString(Namespaces.Utopianism, "guild.dissolve");
        }
    }
}

File: src/main/java/io/myosotisdev/utopianism/command/GuildCommand.kt
package io.myosotisdev.utopianism.command

import io.myosotisdev.minestom.command.AbstractCommand
import io.myosotisdev.minestom.permission.Permission
import io.myosotisdev.utopianism.util.Namespaces

class GuildCommand : AbstractCommand{
    companion object {
        val Name: String = "guild"
        val Aliases: Array<String> = arrayOf()
    }

    constructor(command: AbstractCommand?) : super(command, Name)
    {
        addSubcommand(Create(this))
    }

    override fun permission(): Permission {
        return Permission.ofString(Namespaces.Utopianism, Name)
    }

    class Create(command: AbstractCommand) : AbstractCommand(command, Name)
    {
        companion object {
            const val Name: String = "create"
        }

        init
        {

        }

        override fun permission(): Permission {
            return Permission.ofString(Namespaces.Utopianism, "guild.create")
        }
    }

    class Dissolve(command: AbstractCommand) : AbstractCommand(command, Name)
    {
        companion object { val Name: String = "dissolve" }

        init
        {
            
        }

        override fun permission(): Permission {
            return Permission.ofString(Namespaces.Utopianism, "guild.dissolve")
        }
    }
}